lib/ruby_routes/route/segment_compiler.rb

```
@@ -0,0 +1,68 @@
+      end
+
+      def extract_path_params_fast(request_path)
+        return RubyRoutes::Constant::EMPTY_HASH if @compiled_segments.empty? &&
+                                                   request_path == RubyRoutes::Constant::ROOT_PATH
+        return nil if @compiled_segments.empty?
+        parts = split_path(request_path)
+        with_splat = @compiled_segments.any? { |segment| segment[:type] == :splat }
+        return nil if (!with_splat && parts.size != @compiled_segments.size) ||
+                      (with_splat && parts.size < (@compiled_segments.size - 1))
+        extract_params_from_parts(parts)
+      end
```

RuboCop (1.76.1)

[convention] Cyclomatic complexity for `extract_path_params_fast` is too high. [9/7]

(Metrics/CyclomaticComplexity)